src/main/groovy/com/github/erdi/gradle/webdriver/repository/DriverUrlsConfiguration.groovy

```diff
@@ -52,19 +55,21 @@ class DriverUrlsConfiguration {
             architectures << X86
         }
         def platform = platform(spec.os)
-        def bits = architectures.collect { bit(it) }
+        def bits = architectures.collect { bit(it) } as Set
+        def archs = architectures.collect { arch(it) } as Set
         def versionPattern = ~spec.version
 
         def matchingDrivers = drivers.findAll {
             it.name == spec.name &&
                 it.platform == platform &&
                 versionPattern.matcher(it.version.toString()).matches() &&
-                bits.contains(it.bit)
+                bits.contains(it.bit) &&
+                (it.arch == null || archs.contains(it.arch))
         }.sort { left, right ->
             def leftComparableVersion = new ComparableVersion(left.version.toString())
             def rightComparableVersion = new ComparableVersion(right.version.toString())
 
-            rightComparableVersion <=> leftComparableVersion ?: (right.bit <=> left.bit)
+            rightComparableVersion <=> leftComparableVersion ?: (right.arch <=> left.arch) ?: (right.bit <=> left.bit)
```

Comment:
I also don't think we should have this change here. I don't believe we should ever match on two architectures (apart from matching on both `X86` and `X64_64` which is handled by sorting on bit) so sorting on arch is redundant. I will revert this change.